CSharpMath.Editor.Tests/CaretTests.cs

```diff
@@ -274,4 +221,72 @@ public class CustomizablePlaceholder : IDisposable {
       Assert.Equal(System.Drawing.Color.Blue, LaTeXSettings.Placeholder.Color);
     }
   }
+  [Collection(nameof(NonParallelPlaceholderTests))]
+  public class NonBlinkingPlaceholder : IDisposable {
+    readonly MathKeyboard<TestFont, TGlyph> Keyboard;
+    public NonBlinkingPlaceholder() {
+      LaTeXSettings.PlaceholderBlinks = false;
+      Keyboard = new MathKeyboard<TestFont, TGlyph>(TestTypesettingContexts.Instance, new TestFont()) {
+        InsertionPositionHighlighted = true
+      };
+    }
+    public void Dispose() => LaTeXSettings.PlaceholderBlinks = DefaultPlaceholderSettings.Blinks;
+    void ExpectedAppearance(string expectedOuterNucleus, string expectedInnerNucleus) {
+      var outer = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(Keyboard.MathList));
+      var inner = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(outer.Subscript));
+      Assert.Equal(expectedOuterNucleus, outer.Nucleus);
+      Assert.Equal(expectedInnerNucleus, inner.Nucleus);
+    }
+    [Fact]
+    public async Task PlaceholderDoesNotBlinkAndNoCaretVisible() {
+      Keyboard.KeyPress(MathKeyboardInput.Subscript);
+      Check();
+      await Task.Delay((int)MathKeyboard<TestFont, TGlyph>.DefaultBlinkMilliseconds + CaretBlinks.MillisecondBuffer);
+      Check();
+      await Task.Delay((int)MathKeyboard<TestFont, TGlyph>.DefaultBlinkMilliseconds + CaretBlinks.MillisecondBuffer);
+      Check();
+      void Check() {
```

Comment:
why definition of `Check` is after its using?

Reply:
I changed it. It is something I do often (a rule that I automatically follow), but in this case it adds nothing. In larger structures I call first, then at scrolling down you can see the internals of what is called if you're interested in reading it. If that methods calls other methods, you can scroll down again, etc. Now the "most important things" are at the top, so that the structure of the whole is the first thing you see, details below.

Reply:
Yeah. You can deduce nothing from the name `Check`. If it is a descriptive name then it would have helped.